bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/BuildManager.java

```diff
@@ -739,6 +739,14 @@ public void endTopLevel(boolean needsBuild) {
 		autoBuildJob.build(needsBuild);
 	}
 
+	public void waitForBuild() {
```

Comment:
The method name should be changed to `waitForAutoBuild`, because we don't wait for any other build kind.

Was there any reason why this is moved to BuildManager, if it is used from tests only so far?

Reply:
Note, BuildManager **is not API** and shouldn't be used outside resources / tests

Reply:
@iloveeclipse 
i can't merge until you resolve the review.